src/ProjectTemplates/Shared/ProcessLock.cs

```diff
@@ -24,7 +24,7 @@ public ProcessLock(string name)
 
         public async Task WaitAsync(TimeSpan? timeout = null)
         {
-            timeout ??= TimeSpan.FromMinutes(2);
+            timeout ??= TimeSpan.FromMinutes(20);
```

Comment:
I know, but these used to wait indefinitely, so it's a minor improvement over that.

Reply:
What goes wrong if they only wait 2 minutes?

Reply:
When I added these timeouts here, the build passed a few times successfully on public AzDo, and promptly failed in the internal Azdo. This of course included the time it takes to build \ publish which is being changed here. What I'm going with here is that it's hard to pick a reasonable time, but we also do not want tests to fail just because restore took too long that one time.